guacamole/src/main/webapp/app/manage/controllers/manageSharingProfileController.js

```diff
@@ -0,0 +1,406 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * The controller for editing or creating sharing profiles.
+ */
+angular.module('manage').controller('manageSharingProfileController', ['$scope', '$injector',
+        function manageSharingProfileController($scope, $injector) {
+
+    // Required types
+    var SharingProfile = $injector.get('SharingProfile');
+    var PermissionSet  = $injector.get('PermissionSet');
+
+    // Required services
+    var $location                = $injector.get('$location');
+    var $routeParams             = $injector.get('$routeParams');
+    var authenticationService    = $injector.get('authenticationService');
+    var connectionService        = $injector.get('connectionService');
+    var guacNotification         = $injector.get('guacNotification');
+    var permissionService        = $injector.get('permissionService');
+    var schemaService            = $injector.get('schemaService');
+    var sharingProfileService    = $injector.get('sharingProfileService');
+    var translationStringService = $injector.get('translationStringService');
+
+    /**
+     * An action which can be provided along with the object sent to showStatus
+     * to allow the user to acknowledge (and close) the currently-shown status
+     * dialog.
+     */
+    var ACKNOWLEDGE_ACTION = {
+        name        : "MANAGE_SHARING_PROFILE.ACTION_ACKNOWLEDGE",
+        // Handle action
```

Comment:
I'm not sure this comment adds very much meaning. "Handle action" seems to be implied by the fact that the action callback is being invoked?